python/ray/data/context.py

```diff
@@ -292,7 +294,8 @@ def get_current() -> "DataContext":
                     use_polars=DEFAULT_USE_POLARS,
                     eager_free=DEFAULT_EAGER_FREE,
                     decoding_size_estimation=DEFAULT_DECODING_SIZE_ESTIMATION_ENABLED,
-                    min_parallelism=DEFAULT_MIN_PARALLELISM,
+                    # NOTE: This parameter is deprecated. Use `min_num_blocks` instead.
+                    min_parallelism=DEFAULT_MIN_NUM_BLOCKS,
```

Comment:
Is `DataContext.min_parallelism` used anywhere after these changes? If not, should we just remove it?

Reply:
`DataContext` is developer API, so I am afraid of users may use it. So didn't remove it in the first place. Maybe let's remove it in 2.11?

Reply:
If we're keeping it, let's make it backwards compatible and raise a deprecation warning? With the current implementation, seems like it a no-op and we don't tell the user that

Reply:
User can set the config directly like:

```
DataContext.get_current().min_parallelism = ...
```

It seems to me there's no easy way to throw a deprecation warning here. WDYT?

Reply:
Discussed offline, added deprecation warning in `_autodetect_parallelism` if `DataContext. min_parallelism` is set to a non-default value.